### PR TITLE
feat: Initial commit

### PR DESCRIPTION
**Pull request by mattcompiles**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: node_js
+cache:
+  directories:
+    - node_modules
+notifications:
+  email: false
+node_js:
+  - '8'
```
Comment (markdalgleish) [inline .travis.yml]: Should use `.nvmrc` instead.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,25 @@
...
+
+This package includes the shareable [TSLint](https://palantir.github.io/tslint/) configuration used by [SEEK](https://github.com/seek-oss/).
+
+Note: this config assumes you are using `tsc` in strict mode. 
+
+## Usage
+
+First, install this package and TSLint.
+
+Then create a file named tslint.json with following contents in the root folder of your project:
```
Comment (markdalgleish) [inline README.md]:
```suggestion
Then create a file named `tslint.json` with following contents in the root folder of your project:
```

File: index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      "interface-name": [
+          true,
+          "never-prefix"
+      ],
+      "no-implicit-dependencies": [
+          true,
+          "dev"
+      ]
+  }
+}
```
Comment (markdalgleish) [inline index.js]:
```suggestion
}

```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "url": "https://github.com/seek-oss/tslint-config-seek.git"
+  },
+  "author": "SEEK",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/seek-oss/tslint-config-seek/issues"
+  },
+  "scripts": {
+    "commit": "git-cz",
+    "travis-deploy-once": "travis-deploy-once",
```
Comment (michaeltaranto) [inline package.json]: Given we are only running our pipeline on one version of node we can remove the need for `travis-deploy-once`

File: package.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    "success": false
+  },
+  "config": {
+    "commitizen": {
+      "path": "./node_modules/cz-conventional-changelog"
+    }
+  },
+  "dependencies": {
+    "tslint-config-prettier": "^1.15.0"
+  }
```
Comment (michaeltaranto) [inline package.json:43]: Thoughts on adding `tslint` as a peer dep?
```suggestion
  },
  "peerDependencies": {
    "tslint": ">=5"
  }
```

File: index.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  "extends": [
+      "tslint:latest",
+      "tslint-config-prettier"
+  ],
+  "rules": {
+      "object-literal-sort-keys": false,
```
Comment (akreitals) [inline index.js]: Only other inclusion I would suggest is: `"no-submodule-imports": false`.

Additionally I typically put a link to the rule details i.e.
```
        // https://palantir.github.io/tslint/rules/no-submodule-imports/
        "no-submodule-imports": false,
        // https://palantir.github.io/tslint/rules/object-literal-sort-keys/
        "object-literal-sort-keys": false,
        // https://palantir.github.io/tslint/rules/ordered-imports/
        "ordered-imports": false,
        // https://palantir.github.io/tslint/rules/interface-name/
        "interface-name": [
            true,
            "never-prefix"
        ],
        // https://palantir.github.io/tslint/rules/no-implicit-dependencies/
        "no-implicit-dependencies": [
            true,
            "dev"
        ]
```

Comment (akreitals) [reply index.js]: I should preface that with an example - `lodash/fp`.

I know you can add a whitelist as well but that would be too difficult to manage. I actually can't think of any others off the top of my head so not a strong case for the inclusion here.

Comment (etaoins) [reply index.js]: I’m not opposed to that. I’ve been following no submodule imports in Node but on the browser that’s putting a lot of faith in tree shaking for e.g. lodash.

Comment (mattcompiles) [reply index.js]: Yeah turning that off makes sense. I agree with the rule in general but too many edge cases.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,25 @@
+[![Build Status](https://img.shields.io/travis/seek-oss/tslint-config-seek/master.svg?style=flat-square)](http://travis-ci.org/seek-oss/tslint-config-seek) [![npm](https://img.shields.io/npm/v/tslint-config-seek.svg?style=flat-square)](https://www.npmjs.com/package/tslint-config-seek) [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg?style=flat-square)](https://github.com/semantic-release/semantic-release) [![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg?style=flat-square)](http://commitizen.github.io/cz-cli/)
+
 # tslint-config-seek
-TSLint configuration used by SEEK
+
+This package includes the shareable [TSLint](https://palantir.github.io/tslint/) configuration used by [SEEK](https://github.com/seek-oss/).
+
+Note: this config assumes you are using `tsc` in strict mode.
```
Comment (etaoins) [inline README.md]: > this assumes you’re using a `tsconfig.json` with `strict` and `noUnusedLocals` enabled.

**Pull request merged**